### PR TITLE
Support Arcadia Spark batch submission

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java [hunk 1]
```diff
@@ -104,7 +104,14 @@ public String getConnectionUrl() {
         }
 
         try {
-            return new URIBuilder(this.workSpace.getSparkUrl()).setPort(ARCADIA_SPARK_SERVICE_PORT).build().toString();
+//            return new URIBuilder(this.workSpace.getSparkUrl()).setPort(ARCADIA_SPARK_SERVICE_PORT).build().toString();
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java]: >  return new URIBui [](start = 12, length = 19)

remove this line

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmitModel.kt [hunk 2]
```diff
@@ -23,8 +23,18 @@
 package com.microsoft.azure.hdinsight.spark.run.configuration
 
 import com.intellij.openapi.project.Project
+import com.intellij.util.xmlb.annotations.Attribute
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmitModel
 
 class ArcadiaSparkSubmitModel(project: Project) : SparkSubmitModel(project) {
+    @Attribute("livy_uri")
+    lateinit var livyUri: String
```
Comment (lcadzy) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmitModel.kt]: >String [](start = 26, length = 6)

String? or String

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ArcadiaSparkBatchJob.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package com.microsoft.azure.hdinsight.spark.common;
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ArcadiaSparkBatchJob.java]: copy right missing

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchArcadiaSubmission.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
+package com.microsoft.azure.hdinsight.spark.common;
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchArcadiaSubmission.java]: copy right missing

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/UriUtil.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.microsoft.azure.hdinsight.common;
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/UriUtil.java]: copy right missing

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java (hunk 1 above)
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java]: Let's keep this for what it should be. I'll add comments here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmitModel.kt (hunk 2 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmitModel.kt]: It's `lateinit`, which can be `String`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ArcadiaSparkBatchJob.java (hunk 3 above)
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ArcadiaSparkBatchJob.java]: fix it.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchArcadiaSubmission.java (hunk 4 above)
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchArcadiaSubmission.java]: fix it

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/UriUtil.java (hunk 5 above)
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/UriUtil.java]: fix it

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpResponse.java [hunk 6]
```diff
@@ -21,21 +21,23 @@
...
+    private List<Header> headers;
     private String content;
 
-    public HttpResponse(int code, String message,Map<String, List<String>> headers,
+    public HttpResponse(int code, String message, Header[] headers,
                         String content) {
         this.code = code;
         this.message = message;
-        this.headers = new HashMap<>(headers);
+        this.headers = Arrays.asList(headers);
```
Comment (t-rufang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpResponse.java]: Will headers be null here? If it is, this code will hit NPE error.

Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpResponse.java]: Fix it by adding NPE check here

**Comment by lcadzy** [comment]
Thank you wei! LGTM!

**Pull request merged**